src/serializer.ts

```diff
@@ -62,12 +70,23 @@ class Serializer {
    * @returns {Object} The Payment as JSON.
    */
   private static paymentToJSON(payment: Payment): object | undefined {
-    const json = {
-      TransactionType: "Payment",
-      Destination: payment.getDestination(),
-      Amount: {}
+    const json: PaymentJSON = {
+      Amount: {},
+      Destination: "",
+      DestinationTag: 0,
+      TransactionType: "Payment"
     };
 
+    // If an x-address was able to be decoded, add the components to the json.
+    const decodedXAddress = Utils.decodeXAddress(payment.getDestination());
+    if (decodedXAddress === undefined) {
```

Comment:
NIT: I'm curious why the explicit check on undefined over just `if (!decodedXAddress) {`? Is this just a best practice type of thing? Also curious why the function decodeXAddress returns undefined vs just doing a naked return.

Reply:
I honestly just figured there was some weird behavior in javascript if I used this style :) done.

Reply:
> Also curious why the function decodeXAddress returns undefined vs just doing a naked return.

You're suggesting that x-address would just return the classic address, if a classic address is passed as input. There are 3 cases:


| input  | As Implemented | Naked Return |
|---|---|---|
| valid classic address | undefined | input classic address |
| valid x address | undefined | decoded classic address |
|some invalid string ("xrp") | undefined | undefined (?)|

I think returning undefined is the most expected thing. Naked returns complicate the return values of this function (now 3 things can be returned). 

You could argue that I should really implement an `isXAddress` function, which might help this, but I'd still need to do something like:
```javascript
if (isXAddr(addr) && decodeXAddr(addr) != undefined) {
    // do stuff 
}
```

or 

```javascript 
if (isXAddr(xaddr)) {
   const decoded = decodeXAddr(xAddr);
   // if decoded is undefined, try some error handling
}
```

Reply:
Resolving since Tyler approved changes.